internal/internal_event_handlers.go

```diff
@@ -826,6 +826,8 @@ func (weh *workflowExecutionEventHandlerImpl) ProcessEvent(
 	case enumspb.EVENT_TYPE_WORKFLOW_TASK_STARTED:
 		// Set replay clock.
 		weh.SetCurrentReplayTime(common.TimeValue(event.GetEventTime()))
+		// Set history length as this event's ID
+		weh.workflowInfo.currentHistoryLength = int(event.EventId)
```

Comment:
I don't know if this is sufficient. Based on what's going on here: https://cs.github.com/temporalio/sdk-go/blob/2f76f4de20f0171cb67a6881853c6969f2324761/internal/internal_task_handlers.go#L910

Event ids can be processed out-of-order when looking at markers, which makes sense, but also means this line should probably be more like:
`weh.workflowInfo.currentHistoryLength = max(weh.workflowInfo.currentHistoryLength, int(event.EventId))

Reply:
This is just the event ID of workflow task started. While we do process change the order of history during processing, I believe we only do it one task at a time. So at the top of that loop iteration from your link there is `reorderedHistory.NextCommandEvents()`. Arguably I could just put this value right after we obtain that, but I think the effect is the same.

I may have to re-read the code to confirm, it is a bit hard to follow.